llamafile/server/client.cpp

```diff
@@ -520,14 +520,23 @@ Client::send_response_finish()
 //
 // unlike send() this won't fail if binary content is detected.
 bool
-Client::send_binary(const void* p, size_t n)
-{
-    ssize_t sent;
-    if ((sent = write(fd_, p, n)) != n) {
-        if (sent == -1 && errno != EAGAIN && errno != ECONNRESET)
-            SLOG("write failed %m");
-        close_connection_ = true;
-        return false;
+Client::send_binary(const void* p, size_t n) {
+    const char* buf = (const char*)p;
+    size_t written = 0;
+    while (written < n) {
+        ssize_t sent = write(fd_, buf + written, n - written);
+        if (sent == -1) {
+            if (errno == EAGAIN || errno == EWOULDBLOCK) {
+                // no data can be written right now; retry
+                continue;
```

Comment:
The right thing to do here would be to call poll() I think to wait until it's writable.